aiohttp/client.py

```diff
@@ -414,11 +414,8 @@ def request(
 
     def _build_url(self, str_or_url: StrOrURL) -> URL:
         url = URL(str_or_url)
-        if self._base_url is None:
-            return url
-        else:
-            assert not url.absolute
-            return self._base_url.join(url)
+        need_join = self._base_url and not url.absolute
+        return self._base_url.join(url) if need_join else url
```

Comment:
The PR already meets all your other requirements, but if I put the condition in a named variable, it fails the Linter check.

Linter thinks that `self._base_url` may be `None`, and `None` does not have attribute `join`.
What should I do?

Reply:
Let's try
```suggestion
        need_join = self._base_url and not url.absolute
        return self._base_url.join(url) if need_join else url
```
then. @Dreamsorcerer mentioned type narrowing being imperfect in MyPy..

Reply:
Yeah, that will just result in the variable being a `bool`, it won't narrow the type of `_base_url` in the if block.

Best to drop the variable entirely (avoiding the one-liner should retain some readability):
```suggestion
        if self._base_url and not url.absolute:
            return self._base_url.join(url)
        return url
```